rustls/src/conn.rs

```diff
@@ -731,6 +764,49 @@ impl ConnectionCommon {
     pub(crate) fn read_tls(&mut self, rd: &mut dyn io::Read) -> io::Result<usize> {
         self.message_deframer.read(rd)
     }
+
+    pub(crate) fn export_keying_material(
+        &self,
+        output: &mut [u8],
+        label: &[u8],
+        context: Option<&[u8]>,
+    ) -> Result<(), Error> {
+        match self.state.as_ref() {
+            Ok(st) => st.export_keying_material(output, label, context),
+            Err(e) => Err(e.clone()),
+        }
+    }
+}
+
+#[cfg(feature = "quic")]
```

Comment:
Merges `quic::read_hs()` and its callers in the `QuicExt` implementations. Since this code is fairly tightly coupled to the `ConnectionCommon` and `process_new_handshake_messages()` in particular, this seems like a better interface, at the cost of having more QUIC-specific code in the shared `conn` module. (On the plus side, `ConnectionCommon::process_new_handshake_messages()` is now private.)